grafana-plugin/src/utils/types.ts

```diff
@@ -8,3 +8,7 @@ export interface TableColumn {
   className?: string;
   grow?: number;
 }
+
+export type KeysMatching<T extends object, V> = keyof {
+  [K in keyof T as T[K] extends V ? K : never]: any;
+};
```

Comment:
Works cool! I would suggest more explicit naming though + using `unknown` instead of `any`:
sth like
```
export type PropertiesThatExtendsAnotherClass<OriginalObj, AnotherClass> = keyof {
  [Prop in keyof OriginalObj as OriginalObj[Prop] extends AnotherClass ? Prop : never]: unknown;
};
```

Reply:
fixed